src/kolibri_android/android_whitenoise.py

```diff
@@ -1,88 +1,130 @@
 import logging
-import os
-import re
 import stat
 from urllib.parse import urlparse
-from wsgiref.headers import Headers
 
-from django.contrib.staticfiles import finders
-from django.utils._os import safe_join
+from django.utils.functional import cached_property
+from evil_kolibri.utils.kolibri_whitenoise import compressed_file_extensions
+from evil_kolibri.utils.kolibri_whitenoise import DynamicWhiteNoise
+from evil_kolibri.utils.kolibri_whitenoise import EndRangeStaticFile
+from evil_kolibri.utils.kolibri_whitenoise import FileFinder
 from jnius import autoclass
-from kolibri.utils.kolibri_whitenoise import compressed_file_extensions
-from kolibri.utils.kolibri_whitenoise import EndRangeStaticFile
-from kolibri.utils.kolibri_whitenoise import FileFinder
-from kolibri.utils.kolibri_whitenoise import NOT_FOUND
-from whitenoise import WhiteNoise
 from whitenoise.httpstatus_backport import HTTPStatus
-from whitenoise.responders import MissingFileError
 from whitenoise.responders import NOT_ALLOWED_RESPONSE
 from whitenoise.responders import Response
-from whitenoise.string_utils import decode_path_info
 
 from .android_utils import document_exists
 from .android_utils import document_tree_join
-from .android_utils import get_activity
 from .android_utils import is_document_uri
 from .android_utils import open_file
 from .android_utils import stat_file
 
+# We import from evil_kolibri so we can monkey-patch DynamicWhiteNoise from
+# kolibri without creating a circular dependency.
```

Comment:
I don't see anything that installs an `evil_kolibri` package. Did I miss something?

Reply:
Oops, good catch! Fixed now in the newest version of this commit, with some new stuff added to the Makefile: copying some code from `kolibri` to `evil_kolibri`, and applying a patch. I forgot to commit it earlier :)